src/iface/interface/ipv6.rs

```diff
@@ -288,12 +288,11 @@ impl InterfaceInner {
                         Ipv6OptionFailureType::DiscardSendAll => {
                             return HopByHopResponse::Discard(param_problem());
                         }
-                        Ipv6OptionFailureType::DiscardSendUnicast
-                            if !ipv6_repr.dst_addr.is_multicast() =>
-                        {
-                            return HopByHopResponse::Discard(param_problem());
+                        Ipv6OptionFailureType::DiscardSendUnicast => {
+                            if !ipv6_repr.dst_addr.is_multicast() {
+                                return HopByHopResponse::Discard(param_problem());
+                            }
```

Comment:
we still have to return here if the sender is multicast, to discard the packet, right?

Reply:
Yes, indeed, I misread this in the RFC. I thought only the option needs to be discarded, but like you say, the full packet needs to be discarded, without sending an ICMP parameter problem back.